src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int DEFAULT_RADIUS_CIRCLE = 10;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(5);
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case 0:
+                return new Square(color,random.nextDouble() * 10 + 1);
+            case 1:
+                return new RightTriangle(color, random.nextDouble() * 10 + 1,
+                        random.nextDouble() * 10 + 1);
+            case 2:
+                return new Rectangle(color,random.nextDouble() * 10 + 1,
+                        random.nextDouble() * 10 + 1);
+            case 3:
+                return new IsoscelesTrapezoid(color,random.nextDouble() * 10 + 1,
+                        random.nextDouble() * 10 + 1,
+                        random.nextDouble() * 10 + 1);
+            case 4:
+                return new Circle(color, random.nextDouble() * 10 + 1);
```

Comment:
Instead of passing expressions directly to the constructors, assign the random values to variables first. This will make the code more readable and easier to understand.